src/app/modules/main/wallet_section/send/account_item.nim

```diff
@@ -1,44 +1,126 @@
+    ): AccountItem =
+      new(result, delete)
+      result.setup(name, address, color, emoji, walletType, assets, currencyBalance)
   
-proc `$`*(self: AccountItem): string =
-  result = "WalletSection-Send-Item("
-  result = result & $self.WalletAccountItem
-  result = result & "\nassets: " & $self.assets
-  result = result & "\ncurrencyBalance: " & $self.currencyBalance
-  result = result & ")"
-    
-proc assets*(self: AccountItem): token_model.Model =
-  return self.assets
-
-proc currencyBalance*(self: AccountItem): CurrencyAmount =
-  return self.currencyBalance
+  proc `$`*(self: AccountItem): string =
+    result = "WalletSection-Send-Item("
+    result = result & "\nname: " & $self.name
+    result = result & "\naddress: " & $self.address
+    result = result & "\ncolor: " & $self.color
+    result = result & "\nemoji: " & $self.emoji
+    result = result & "\nwalletType: " & $self.walletType
+    result = result & "\nassets: " & $self.assets
+    result = result & "\ncurrencyBalance: " & $self.currencyBalance
+    result = result & ")"
+
+  proc nameChanged*(self: AccountItem) {.signal.}
+  proc getName*(self: AccountItem): string {.slot.} =
+    return self.name
+  proc setName*(self: AccountItem, value: string) {.slot.} =
+    self.name = value
+    self.nameChanged()
+  QtProperty[string] name:
+    read = getName
+    notify = nameChanged
+
+  proc addressChanged*(self: AccountItem) {.signal.}
+  proc getAddress*(self: AccountItem): string {.slot.} =
+    return self.address
+  proc setAddress*(self: AccountItem, value: string) {.slot.} =
+    self.address = value
+    self.addressChanged()
+  QtProperty[string] address:
+    read = getAddress
+    notify = addressChanged
+
+  proc colorChanged*(self: AccountItem) {.signal.}
+  proc getColor*(self: AccountItem): string {.slot.} =
+    return self.color
+  proc setColor*(self: AccountItem, value: string) {.slot.} =
+    self.color = value
+    self.colorChanged()
+  QtProperty[string] color:
+    read = getColor
+    notify = colorChanged
+
+  proc emojiChanged*(self: AccountItem) {.signal.}
+  proc getEmoji*(self: AccountItem): string {.slot.} =
+    return self.emoji
+  proc setEmoji*(self: AccountItem, value: string) {.slot.} =
+    self.emoji = value
+    self.emojiChanged()
+  QtProperty[string] emoji:
+    read = getEmoji
+    notify = emojiChanged
+
+  proc walletTypeChanged*(self: AccountItem) {.signal.}
+  proc getWalletType*(self: AccountItem): string {.slot.} =
+    return self.walletType
+  proc setWalletType*(self: AccountItem, value: string) {.slot.} =
+    self.walletType = value
+    self.walletTypeChanged()
+  QtProperty[string] walletType:
+    read = getWalletType
+    notify = walletTypeChanged
+
+  proc assetsChanged*(self: AccountItem) {.signal.}
+  proc getAssets*(self: AccountItem): token_model.Model =
+    return self.assets
+  proc getAssetsAsQVariant*(self: AccountItem): QVariant {.slot.} =
+    return newQVariant(self.assets)
```

Comment:
@Khushboo-dev-cpp `self.assets` may be `nil` as well, please use 
```
if self.assets.isNil:
  return newQVariant()
return newQVariant(self.assets)
```

Reply:
in this specific context how can it be nil?

Reply:
getAssets can be right? in case nil is assigned on initialization?

Reply:
I am not sure, i need to double check but as a minimum it would be an empty seq

Reply:
> I am not sure, i need to double check but as a minimum it would be an empty seq

It's a model, so will be either an instance of a model or nil. (Items within that model may or may not be an empty seq)